kernel/kernel-api/src/test/java/io/delta/kernel/internal/types/JsonHandlerTestImpl.java

```diff
@@ -279,13 +286,109 @@ public Row getStruct(int ordinal) {
         }
 
         @Override
-        public <T> List<T> getArray(int ordinal) {
-            return (List<T>) parsedValues[ordinal];
+        public ArrayValue getArray(int ordinal) {
+            return (ArrayValue) parsedValues[ordinal];
         }
 
         @Override
         public <K, V> Map<K, V> getMap(int ordinal) {
             return (Map<K, V>) parsedValues[ordinal];
         }
     }
+
+    private static class TestColumnVector implements ColumnVector {
```

Comment:
Wondering if there is another alternative to this. Does Spark has similar requirement where they need to expose a list as Vector?